addons/website_event_exhibitor/static/src/js/event_sponsor_search.js

```diff
@@ -0,0 +1,49 @@
+odoo.define('website_event_exhibitor.event_sponsor_search', function (require) {
+'use strict';
+
+const publicWidget = require('web.public.widget');
+publicWidget.registry.websiteEventSearchSponsor = publicWidget.Widget.extend({
+
+    selector: '.o_wesponsor_index',
+    events: {
+        'click .o_wevent_event_search_box .btn': '_onSearch',
+        'click .o_search_tag .btn': '_onTagRemove',
+        'click .o_dropdown_reset_tags': '_onTagReset',
+        'change .o_wevent_event_tags_form input': '_onTagAdd',
+    },
+
+    start: function () {
+        this.form = this.$el.find('.o_wevent_event_tags_form');
+        return this._super.apply(this, arguments);
+    },
+
+    _onTagAdd: function () {
+        this.form.submit();
+    },
+
+    _onTagRemove: function (event) {
+        const tag = $(event.target).parent();
+        const data = tag.data();
+        const selector = 'input[name="' + data.field + '"][value="' + data.value + '"]';
+        this.form.find(selector).prop('checked', false);
+        this.form.submit();
+    },
+
+    _onTagReset: function (event) {
+        const dropdown = $(event.target).parent();
+        dropdown.find('input').prop('checked', false);
+        this.form.submit();
+    },
+
+    _onSearch: function () {
```

Comment:
Linting: onSearch should be defined before onTagAdd (alphabetical ordering in JS)